### PR TITLE
TpetraTSQR: Fix #6354; refactor TSQR

### LINKED ISSUE #6354: Tpetra: compilation error with Tpetra deprecated code off in Tpetra::TsqrAdapter
Along with @mperego, I'm trying to switch the Albany nightlies to turn off the Tpetra/Xpetra deprecated code:

```
 -D Tpetra_ENABLE_DEPRECATED_CODE:BOOL=OFF \
 -D Xpetra_ENABLE_DEPRECATED_CODE:BOOL=OFF \
```

I encountered the following compilation error in most of my nightlies when doing this: 

```
repos/Trilinos/packages/tpetra/core/src/Tpetra_TsqrAdaptor.hpp:353:72: error: no member named 'getNode' in 'Tpetra::Map >'
```

http://cdash.sandia.gov/CDash-2-3-0/viewBuildError.php?buildid=91235 

A sample configuration that produces the error can be found here: https://github.com/SNLComputation/Albany/blob/master/doc/dashboards/cee-compute011.sandia.gov/ctest_nightly.cmake

### PR DESCRIPTION
**Pull request by mhoemmen**
@trilinos/tpetra 

## Motivation

1. Fix build error with deprecated code off (#6354).
2. Refactor TSQR for eventual use of Kokkos and TPLs (cuSOLVER in particular).

## Related Issues

* Closes #6354 

## Stakeholder Feedback

Feedback from Albany tests; see #6354 discussion.

### CODE REVIEW

File: packages/tpetra/tsqr/src/Tsqr_KokkosNodeTsqr.hpp [hunk 1]
```diff
@@ -1199,85 +1112,16 @@ namespace TSQR {
...
-      node_ = node;
-    }
-
     /// \brief Whether this object is ready to perform computations.
-    ///
-    /// It is <i>not</i> ready if the Kokkos Node instance has not yet
-    /// been set.
     bool ready() const {
-      return ! getNode().is_null();
+      return true;
```
Comment (mperego) [inline packages/tpetra/tsqr/src/Tsqr_KokkosNodeTsqr.hpp]: Why is this always true?

Comment (mhoemmen) [reply packages/tpetra/tsqr/src/Tsqr_KokkosNodeTsqr.hpp]: @mperego The class no longer depends on a Kokkos 1.0 Node object.  It used to be that the class wasn't "ready" until somebody called it with a `nonnull RCP<Node>`.

**Comment by mhoemmen** [comment]
My bad -- I'll fix this one today.

**Comment by mhoemmen** [comment]
@mperego @ikalash Sorry folks -- there was just one build error for when complex Scalar types were turned off.  I pushed a fix for that.

**Pull request merged**